addons/sale/models/sale_order_line.py

```diff
@@ -264,11 +264,16 @@ class SaleOrderLine(models.Model):
 
     #=== COMPUTE METHODS ===#
 
-    @api.depends('order_partner_id', 'order_id', 'product_id')
+    @api.depends('order_partner_id', 'order_id', 'product_id', 'is_downpayment')
     def _compute_display_name(self):
         name_per_id = self._additional_name_per_id()
         for so_line in self.sudo():
-            name = '{} - {}'.format(so_line.order_id.name, so_line.name and so_line.name.split('\n')[0] or so_line.product_id.name)
+            name = '{} - {}'.format(
+                so_line.order_id.name,
+                so_line.name and so_line.name.split('\n')[0] or (
+                    _('Down Payment') if so_line.is_downpayment else so_line.product_id.name
+                )
```

Comment:
AFAIK, so lines always have a name (dp lines too), why do you need this fallback ? :thinking:

Reply:
IMHO, either we do not change anything here, or we remove the fallback on the product name.

Anyway, for dp lines, we know the name will be set and correctly computed through the compute method.
(and I don't think we have to support the case when customers empties the name field on a SO line).